File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int extra) {
      priority_queue<int,vector<int>,greater<int>>left;
      for(int i=0; i<rocks.size(); i++)
        left.push(capacity[i]-rocks[i]);
      int ret = 0;
      while(left.size() && extra > 0){
        if(left.top() > extra){
          return ret;
        }
        if(left.top() <= extra){
          extra -= left.top();
          left.pop();
          ret++;
        }
      }
      return ret;
    }
};

File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.py
class Solution:
    def maximumBags(self, cap: List[int], rocks: List[int], extra: int) -> int:
      left = [cap[i] - rocks[i] for i in range(len(cap))]
      left.sort()
      ret = 0
      for n in left:
        extra -= n
        if(extra < 0):
          return ret
        ret += 1
      return ret;
        